code/modules/economy/store/store_credits.dm

```diff
@@ -115,8 +126,10 @@
 /obj/machinery/store/proc/buy_and_transfer()
 	var/list/things = list() + buy_current()
 	var/obj/item/weapon/rig/R
-	for (var/obj/item/I in things)
-		if (isrig(I))
+	if(occupant.back && !occupant.wearing_rig)
+		store_or_drop(occupant.back)
+	for(var/obj/item/I in things)
+		if(isrig(I))
 			R = I
```

Comment:
Just use `var/obj/item/weapon/rig/R = locate(/obj/item/weapon/rig) in things` instead of for loop

Reply:
There is `store_or_drop(I)` called for every item in `things`, although I'm not sure if there is a situation when that list contain more than one object.

Reply:
In that case use `locate()` then call `start_transfer(R)` and remove it from `things`, after that we can loop through all items in `things` without checking if it's a RIG.